RUFAS/biophysical/manure/handler.py

```diff
@@ -0,0 +1,281 @@
+        """
+        Implements the basic checks for receiving manure stream.
+
+        Parameters
+        ----------
+        manure_stream : ManureStream
+            The ManureStream instance being checked for compatibility.
+
+        """
+        info_map = {"class": self.__class__.__name__, "function": self.receive_manure.__name__}
+        if not self.check_manure_stream_compatibility(manure_stream):
+            self._om.add_error(
+                "Invalid manure stream.",
+                "Received manure stream is not compatible with a handler type processor.",
+                info_map,
+            )
+            raise ValueError("ValueError: Invalid manure stream for handler processor.")
+
+    def process_manure(self, conditions: CurrentDayConditions, time: Time) -> dict[str, ManureStream]:
+        """
+        Executes the daily manure processing operations.
+
+        Parameters
+        ----------
+        conditions : CurrentDayConditions
+            Current weather and environmental conditions that manure is being processed in.
+        time : Time
+            Time instance containing the simulations temporal information.
+
+        Returns
+        -------
+        dict[str, ManureStream]
+            Mapping between classification of manure coming out of this processor to the ManureStream containing the
+            manure information. If the processor is a separator, the classifications are "solid" and "liquid". Otherwise
+            the only classification is "manure".
+
+        """
+        if self.manure_stream is None or self.manure_stream.pen_manure_data is None:
+            info_map = {"class": Handler.__class__.__name__, "function": Handler.process_manure.__name__}
+            self._om.add_error(
+                "None type ManureStream.",
+                "The processed ManureStream or pen data of the manure stream is None type.",
+                info_map,
+            )
+            raise TypeError("TypeError: Handler tries to process 'NoneType' object ManureStream.")
+
+        info_map_c = {"units": MeasurementUnits.DEGREES_CELSIUS}
+        info_map_m3 = {"units": MeasurementUnits.CUBIC_METERS}
+        cleaning_water_volume = self.determine_cleaning_water_volume_in_main_barn(
+            self.manure_stream.pen_manure_data.num_animals,
+            self.config.cleaning_water_use_rate,
+            self.config.cleaning_water_recycle_fraction,
+        )
+        barn_temperature = self.determine_barn_temperature(conditions.mean_air_temperature)
+
+        total_cleaning_water_volume = (
+            cleaning_water_volume + self.fresh_water_volume_used_for_milking
+        ) * GeneralConstants.LITERS_TO_CUBIC_METERS
+        self._om.add_variable("total_cleaning_water_volume", total_cleaning_water_volume, info_map_m3)
+        self._om.add_variable("barn_temperature", barn_temperature, info_map_c)
+
+        manure_water = self.manure_stream.water + cleaning_water_volume
```

Comment:
```suggestion
        manure_water = self.manure_stream.water + (total_cleaning_water_volume * GeneralConstants.WATER_DENSITY_KG_PER_M3)
```
I had this wrong in the design doc, my apologies for that. I am updating this in the design doc now. Water in ManureStream is in kg, so added the density conversion constant here.